crates/matrix-sdk-ui/src/timeline/tests/basic.rs

```diff
@@ -175,15 +177,16 @@ async fn test_other_state() {
         .handle_live_state_event(&ALICE, RoomNameEventContent::new("Alice's room".to_owned()), None)
         .await;
 
-    let _day_divider = assert_next_matches!(stream, VectorDiff::PushBack { value } => value);
-
     let item = assert_next_matches!(stream, VectorDiff::PushBack { value } => value);
     assert_let!(TimelineItemContent::OtherState(ev) = item.as_event().unwrap().content());
     assert_let!(AnyOtherFullStateEventContent::RoomName(full_content) = ev.content());
     assert_let!(FullStateEventContent::Original { content, prev_content } = full_content);
     assert_eq!(content.name, "Alice's room");
     assert_matches!(prev_content, None);
 
+    let day_divider = assert_next_matches!(stream, VectorDiff::Insert { index: 0, value } => value);
+    assert!(day_divider.is_day_divider());
```

Comment:
Is the fact this this assertion moved evidence that a bug was fixed? If so, I'd really like single test that makes it clear what the bug was and that it was fixed. (I.e. one that used to fail and now passes.)

Reply:
Same here, it's only the day divider is added to the array *after* the event has been received.